.github/workflows/release.yaml

```diff
@@ -1,83 +1,131 @@
+          export BASE_ARTIFACTS_VERSIONED_DIR="$(dirname ${BASE_ARTIFACTS_SEARCH_DIR})/${ARTIFACT_VERSIONED_PACKAGE_NAME}" # Regardless of the dir type, relative or absolute
+
+          # ex: @axelarnetwork/interchain-token-service
+          echo "PACKAGE_NAME=${PACKAGE_NAME}" >> $GITHUB_ENV
+
+          # ex: v2.1.0
+          echo "PUBLISHED_PACKAGE_VERSION=${PUBLISHED_PACKAGE_VERSION}" >> $GITHUB_ENV
+
+          # ex: ./artifacts
+          echo "BASE_ARTIFACTS_SEARCH_DIR=${BASE_ARTIFACTS_SEARCH_DIR}" >> $GITHUB_ENV
+
+          # ex: interchain-token-service-assets-v2.1.0
+          echo "ARTIFACT_VERSIONED_PACKAGE_NAME=${ARTIFACT_VERSIONED_PACKAGE_NAME}" >> $GITHUB_ENV
+
+          # ex: ./interchain-token-service-assets-v2.1.0
+          echo "BASE_ARTIFACTS_VERSIONED_DIR=${BASE_ARTIFACTS_VERSIONED_DIR}" >> $GITHUB_ENV
 
-      - name: Generate release
+      # Create `zip` and `tar` archive files for contracts ABIs
+      - name: Create `zip` and `tar` archive files for contracts ABIs
         run: |
-          new_release_branch=""
-
-          if [ "${{ github.ref }}" = "refs/heads/main" ]; then
-            if [ "${{ github.event.inputs.bumpType }}" = "major" ]; then
-              let new_major=${{ steps.parsed-release.outputs.major }}+1
-              new_release_branch="releases/$new_major.0.x"
-            elif [ "${{ github.event.inputs.bumpType }}" = "minor" ]; then
-              let new_minor=${{ steps.parsed-release.outputs.minor }}+1
-              new_release_branch="releases/${{ steps.parsed-release.outputs.major }}.$new_minor.x"
-            else
-              echo "cannot make patch release from main branch"
-              exit 1
-            fi
-          elif [[ "${{ github.ref }}" = refs/heads/releases/* ]]; then
-            if [ "${{ github.event.inputs.bumpType }}" != "patch" ]; then
-              echo "cannot make major/minor release from release branch"
-              exit 1
-            fi
-          else
-            echo "can only release from main or releases/* branches"
-            exit 1
-          fi
-
-          git config --global user.email "cicd@axelar.network"
-          git config --global user.name "axelar-cicd-bot"
-
-          git commit --allow-empty -m "${{ steps.release.outputs.new_tag }}"
-          git push
-
-          if [ -n "$new_release_branch" ]; then
-            git checkout -b $new_release_branch
-            git push -u origin $new_release_branch
-          fi
-
-      - name: Bump version and push tag
-        uses: anothrNick/github-tag-action@1.39.0
+          # Renaming to keep the dir structure in the archive matching the release version defined in `BASE_ARTIFACTS_VERSIONED_DIR`
+          mv ${{ env.BASE_ARTIFACTS_SEARCH_DIR }} ${{ env.BASE_ARTIFACTS_VERSIONED_DIR }}
+
+          find "${{ env.BASE_ARTIFACTS_VERSIONED_DIR }}/contracts" -type f -name "*.json" ! -name "*.dbg.json" -print | zip ${{ env.ARTIFACT_VERSIONED_PACKAGE_NAME }}.zip -@
```

Comment:
`npm clean:artifacts` deletes *.dbg.json already

Reply:
yeah, should avoid duplicating, the npm task should be responsible building artifacts appropriately